### PR TITLE
Add optional parameters when building an image

### PR DESCRIPTION
**Pull request by Stevent-fei**
### Describe what this PR does / why we need it

Add new flag `--downloadImage` when for command `sealer build`.

The effect of this command: When --downloadImage is true, the image of manifests and charts package will be cached, and when it is false, it will not be cached,the default is true

### Describe how you did it

sealer build -f Kubefile -t test:1.1 --downloadImage=false .

### CODE REVIEW

**Comment by justadogistaken** [comment]
I assume that the ClusterImage contains all static files for a user-define k8s. If we allow the "downloadImage", there will be cluster images that can't handle application distribution in special environment.

**Comment by kakaZhou719** [comment]
> I assume that the ClusterImage contains all static files for a user-define k8s. If we allow the "downloadImage", there will be cluster images that can't handle application distribution in special environment.

In some case, users do not need to download the container image in advance. they just want to use sealer to startup the cluster and the node could access the external network to launch pod.
so, i thought we could support to build a clusterimage without application  container image.

File: cmd/sealer/cmd/build.go [hunk 1]
```diff
@@ -95,8 +97,9 @@ func init() {
 	buildCmd.Flags().StringVarP(&buildConfig.BuildType, "mode", "m", "lite", "ClusterImage build type, default is lite")
 	buildCmd.Flags().StringVarP(&buildConfig.KubefileName, "kubefile", "f", "Kubefile", "Kubefile filepath")
 	buildCmd.Flags().StringVarP(&buildConfig.ImageName, "imageName", "t", "", "the name of ClusterImage")
+	buildCmd.Flags().BoolVar(&buildConfig.DownloadImage, "downloadImage", true, "download manifest image")
```
Comment (kakaZhou719) [inline cmd/sealer/cmd/build.go:100]: we need more clear parameter description here：build with application container image, default value is true.

Comment (kakaZhou719) [inline cmd/sealer/cmd/build.go:100]: rename “downloadImage” to “save-app-image”？

**Comment by justadogistaken** [comment]
> > I assume that the ClusterImage contains all static files for a user-define k8s. If we allow the "downloadImage", there will be cluster images that can't handle application distribution in special environment.
> 
> In some case, users do not need to download the container image in advance. they just want to use sealer to startup the cluster and the node could access the external network to launch pod. so, i thought we could support to build a clusterimage without application container image.

Yes, I understand the feature, but I mean if this is allowed. There will exist online cluster-image over registies, it may have problem. I know we don't have the problem for base image distribution now, but it may left a issue for the future.

@bxy4543 @allencloud @VinceCui How do you like this feature?

**Comment by allencloud** [comment]
After the discussion of sealer maintainer meeting, this is a potential feature in the future. While when sealer has been refactored in image spec field and Kubefile field.

Actually it is related to the tradeoff of image consistency and flexibility. Attach the `Holding` label.

**Pull request closed without merging**